### PR TITLE
Include placeholder option on selects if include_blank_option

### PR DESCRIPTION
**Pull request by drusepth**
When the `include_blank_option` option is set on a field, we currently prepend an empty `<option></opton>` to the `<select>`. This, however, results in an empty select by default rather than having any kind of placeholder text. See https://github.com/dobtco/screendoor-v2/issues/4139 for the proposed fix, implemented below.

These screenshots are from this library's kitchen sink (`examples/index.html`). I still need to include additional translations and publish a new version to rails-assets.

Before:

2017-06-27-165723_237x164_scrot

After:

2017-06-27-165737_274x167_scrot

The default option used to give the placeholder is also hidden from the dropdown:

2017-06-27-165826_289x168_scrot

...and cannot be selected again even after selecting another option:

2017-06-27-165834_245x166_scrot

### CODE REVIEW

**Comment by vlymar** [comment]
@jrubenoff, what is our current stance on translations? Does https://github.com/dobtco/dobt/issues/2159#issuecomment-294989205 apply here?

**Comment by vlymar** [comment]
To clarify, can we just add the new string to en.yml and not deal with phraseapp?

**Comment by jrubenoff** [comment]
We should ask for additional translations now, given that we currently support them, and revisit the comment you linked to in the future.

**Comment by vlymar** [comment]
@drusepth let me know when this is ready for review

**Comment by vlymar** [comment]
> Warning: only run this from the master branch.
https://github.com/dobtco/formrenderer-base#to-push-translations-to-phraseapp

If I'm reading this right, we'd merge this PR before generating translations anyway...so I think we can separate the phraseapp process into a followup PR.

**Comment by vlymar** [comment]
Your approach looks good to me.

I wonder if there's any argument to be made for leaving the placeholder option there as a disabled but visible option? Any input @jrubenoff 

I also noticed this seems to be the behavior in Safari, regardless of the `display: none` class. Chrome and FF do disappear that option as intended.

**Safari**
safari_dropdown

**Chrome**
chrome_dropdown

**FF**
ff_dropdown

File: dist/styles/base.scss [hunk 1]
```diff
@@ -248,6 +248,11 @@
   }
 }
 
+// Include a "hidden option" class that can be placed on empty placeholder options to keep them out of the dropdown
+.hidden_option {
```
Comment (vlymar) [inline dist/styles/base.scss]: I noticed all the other classes are prefixed with `fr_`, guessing thats a convention we might want to follow here.

Also, this is super minor, and I could be totally wrong, but I think we try to stick with semantic class naming. My guess is that `placeholder_option` might be more semantic than `hidden_option`. I usually think of semantic as "what is it" vs "how does it look"

**Comment by vlymar** [comment]
besides the css naming nitpick and the question over whether this should be hidden at all, this lgtm.

my next question is how should this be tested?

File: dist/styles/base.scss (hunk 1 above)
Comment (drusepth) [inline dist/styles/base.scss]: Agreed on both of those. I'll update the name with prefix -- I would guess `fr_` is the prefix for formrenderer, so good call.

**Comment by drusepth** [comment]
Thanks, I updated the CSS name and I'm open to whether the option should be hidden or not (@jrubenoff do you have thoughts on that?). I assumed not just to mimic placeholder behavior as closely as possible, but looking around [1][2] it doesn't seem like Safari allows anything similar to `display: none` on options, with the exception of perhaps

```
.fr_placeholder_option {
  position: absolute;
  height: 0;
  width: 0;
}
```

which supposedly should work (I haven't tested yet), but smells extra hacky to me.

As for testing, I would propose the following flow -- suggestions welcome:

1. Ensure this fix works properly standalone in formrenderer-base, and merge when the functionality/looks looks good in the kitchen sink (examples/index.html)
2. Update Screendoor's Gemfile to lock `rails-assets-formrenderer-base` to version 1.0.0 (the current version)
3. Publish these changes with a new version to rails-assets
4. I'll submit a PR removing the gem version lock in Screendoor (so it builds using the latest version) so we can ensure the changes look good in Screendoor before switching over to the new formrenderer-base version

That seems safest to ensure nothing changes for users, but may be overkill since it's such a minor display bug. We can also just merge this, publish to rails-assets, and test locally and/or on staging with a push (which should build the latest gems) to make sure things look good before we push to production (which would rebuild and update the rails-asset gem).

**Comment by jrubenoff** [comment]
We should always show the default option. That way, for optional questions, respondents can undo their mistake if they accidentally change the dropdown's value.

Also, you can't really style `option` elements at all, even with the `display` property.

**Comment by drusepth** [comment]
Good call. I removed the `option` styling and went ahead and removed the `disabled` property on the element as well, so respondents can undo their mistake if they accidentally change the dropdown's value. 

Looks like I've got the Phraseapp credentials now as well, so as soon as this is merged I'll push and pull translations from master also. 

@vlymar Ready for you to look at again.

**Comment by vlymar** [comment]
@drusepth, do you need to rebuild + commit your changes? (`grunt all`)

**Comment by drusepth** [comment]
Looks like removing the `disabled` properly was keeping Rivets from defaulting to the first element, but adding `value=""` allowed the default selection again.

Fix is up, assets are built, and default `option` is no longer disabled.

2017-06-30-165204_290x161_scrot

2017-06-30-165405_244x220_scrot

Ready for review.

**Comment by vlymar** [comment]
I think we can accelerate the deploy process for this.

My suggestion would be to skip the Screendoor Gemfile locking step and just publish these changes to rails assets. Then we'll deploy screendoor to staging with the newest assets, if that looks good we'll deploy to production.

Something I'm confused about: `src/version.coffee` lists our version # as 1.0, but the most recent tag is 1.1.0.rc1, and thats what i see on [rails-assets](https://rails-assets.org/#/components/formrenderer-base) too

**Comment by vlymar** [comment]
besides versioning, everything else lgtm

**Comment by vlymar** [comment]
also pinged @jrubenoff to do a final pass over this

**Comment by jrubenoff** [comment]
LGTM. When you test in Screendoor, please make sure that blank dropdown fields are submitted as blank (i.e. not with "Choose an option" as the answer.)

**Comment by drusepth** [comment]
I agree on skipping the Gemfile locking step. We can deploy this to staging then as soon as it's merged.

Regarding versioning: it looks like we have two purposes for the versions:

* Major versions (previously 0, now 1) [updated this way](https://github.com/dobtco/formrenderer-base/commit/80fe6ec9d8e3e95cc10eea03ac4d49af0fe97940#diff-a35864fcc4749a3b904641b4be9feacf) are referenced when including the assets manually, e.g. (from the readme):
```
<script src='//d3q1ytufopwvkq.cloudfront.net/1/formrenderer.js'></script>
<link href='//d3q1ytufopwvkq.cloudfront.net/1/formrenderer.css' rel='stylesheet' />
```

* Minor versions (previously 1.0.0, currently on 1.1.0.rc1) seem to just be for our tagging benefit here and on rails-assets -- looks like rails-assets always builds the latest version.

With the above in mind, I recommend we bump the version to 1.1.0 (release it, including everything in rc1) and tag it here as 1.1.0. Rails-assets should tag it there as 1.1.0 as well, and it should be included in the Gemfile as the latest. I pushed up the version bump in https://github.com/dobtco/formrenderer-base/pull/142/commits/7c8466a978f0bb8a9504ab4b3a5c2a2b090a6b7c

**Comment by vlymar** [comment]
sounds good to me

**Pull request merged**